src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3821,6 +3821,57 @@ internal new NamedTypeSymbol CreateNativeIntegerTypeSymbol(bool signed)
             return this.AnonymousTypeManager.ConstructAnonymousTypeSymbol(descriptor).GetPublicSymbol();
         }
 
+        protected override IMethodSymbol CommonCreateBuiltinOperator(
+            string name,
+            ITypeSymbol returnType,
+            ITypeSymbol leftType,
+            ITypeSymbol rightType,
+            bool isChecked)
+        {
+            var csharpReturnType = returnType.EnsureCSharpSymbolOrNull(nameof(returnType));
+            var csharpLeftType = leftType.EnsureCSharpSymbolOrNull(nameof(leftType));
+            var csharpRightType = rightType.EnsureCSharpSymbolOrNull(nameof(rightType));
+
+            // caller already checked all of these were not null.
+            Debug.Assert(csharpReturnType is not null);
+            Debug.Assert(csharpLeftType is not null);
+            Debug.Assert(csharpRightType is not null);
+
+            var syntaxKind = SyntaxFacts.GetOperatorKind(name);
+            if (syntaxKind == SyntaxKind.None)
+                throw new ArgumentException($"Illegal operator name '{name}'", nameof(name));
+
+            var binaryOperatorName = OperatorFacts.BinaryOperatorNameFromSyntaxKindIfAny(syntaxKind, isChecked);
+            if (binaryOperatorName != name)
+                throw new ArgumentException($"'{name}' was not a valid binary operator name along with isChecked={isChecked}", nameof(name));
```

Comment:
Very neat!
This is so much better than my suggestion :)

Reply:
thanks :)